### DISCUSSION TITLE
Push request sample for .Net Framework?

### DISCUSSION THREAD

**Discussion by clindgren-sym** [tomasfabian/ksqlDB.RestApi.Client-DotNet#22] [General]
Hi, Tomas.  I am really impressed with the library you've created.  I would like to use it for a project, but I need to do push queries and my existing project is .Net Framework 4.7.2.  I didn't see an example of a push query in the ksqlDB.RestApi.Client.DotNetFramework.Sample - I got a push query working using the ksqlDB.Api.Client.Samples project, but it uses C# 8.0 features that I don't have access to in my project.  Would you consider adding a push query example to the ksqlDB.RestApi.Client.DotNetFramework.Sample project?  Thank you.

**Comment by tomasfabian** [comment]
HI @RobotGuy22,
thank you very much for your appreciation!

I checked and tried the .NET Framework sample project and there is a push query example. I slightly modified the project and pushed a [sln file](https://github.com/tomasfabian/ksqlDB.RestApi.Client-DotNet/blob/main/Samples/ksqlDB.RestApi.Client.DotNetFramework.Sample/ksqlDB.RestApi.Client.DotNetFramework.Sample.sln) for it. I also changed locally the target framework from net4.8 to net4.72 and it worked, too. 

Could you please let me know what exactly are you missing in the .NET Framework [example program](https://github.com/tomasfabian/ksqlDB.RestApi.Client-DotNet/blob/main/Samples/ksqlDB.RestApi.Client.DotNetFramework.Sample/Program.cs#L56), please? 

Regards Tomas

**Comment by clindgren-sym** [comment]
Hi Tomas.  Thank you so much for the quick response.  I apologize for not understanding your .Net Framework sample better.  The only query I saw in Program.cs was PullQueryExample().  In my working (C# 8.0) push query there is an occurrence of ‘await using’ and ‘await foreach’ which I wasn’t able to successfully translate back to C# 7.3.

Would you please point me at the push query in the sample?  Is it the CreateQuery<>() on line 56?  If so, I’d like to think I can figure it out from there.  Thanks again.

Chuck

**Reply by tomasfabian** [reply]
> Hi Tomas. Thank you so much for the quick response. I apologize for not understanding your .Net Framework sample better. The only query I saw in Program.cs was PullQueryExample(). In my working (C# 8.0) push query there is an occurrence of ‘await using’ and ‘await foreach’ which I wasn’t able to successfully translate back to C# 7.3. Would you please point me at the push query in the sample? Is it the CreateQuery<>() on line 56? If so, I’d like to think I can figure it out from there. Thanks again. Chuck 
Hi @RobotGuy22,

Yes the push query starts on line 56.

**Reply by clindgren-sym** [reply]
Hi Tomas.  Thanks for updating the .Net Framework sample.  I pulled down a zip of it but unfortunately I'm having a problem with the updated version of the .Net Framework sample solution.  On line 6 of KSqlDbRestApiProvider.cs I get an error "The type or namespace name 'Http' does not exist in the namespace 'ksqlDB.Api.Client.Samples' (are you missing an assembly reference?)".  Is it possible there is a missing commit?

**Comment by tomasfabian** [comment]
Hi @RobotGuy22, 
indeed I apologize. I pushed the missing commits.
